src/main/java/com/basho/riak/client/core/operations/TimeSeriesQueryOperation.java

```diff
@@ -0,0 +1,130 @@
+package com.basho.riak.client.core.operations;
+
+import com.basho.riak.client.core.FutureOperation;
+import com.basho.riak.client.core.RiakMessage;
+import com.basho.riak.client.core.converters.TimeSeriesConverter;
+import com.basho.riak.client.core.query.timeseries.QueryResult;
+import com.basho.riak.client.core.util.BinaryValue;
+import com.basho.riak.protobuf.RiakKvPB;
+import com.basho.riak.protobuf.RiakMessageCodes;
+import com.basho.riak.protobuf.RiakPB;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.3
+ */
+public class TimeSeriesQueryOperation extends FutureOperation<QueryResult, RiakKvPB.TsQueryResp, BinaryValue> {
+
+    private static final Logger logger = LoggerFactory.getLogger(TimeSeriesQueryOperation.class);
+
+    private final RiakKvPB.TsQueryReq.Builder reqBuilder;
+
+    private TimeSeriesQueryOperation(Builder builder)
+    {
+        this.reqBuilder = RiakKvPB.TsQueryReq.newBuilder().setQuery(builder.interpolationBuilder);
+    }
+
+    @Override
+    protected QueryResult convert(List<RiakKvPB.TsQueryResp> responses)
+    {
+        // This is not a streaming op, there will only be one response
+        if (responses.size() > 1)
+        {
+            logger.error("Received {} responses when only one was expected.", responses.size());
```

Comment:
to manage log reader's expectations it is better to mention what will happen with the extra responses